custom_components/omnik_inverter/sensor.py

```diff
@@ -159,36 +159,35 @@ def update(self):
             nextValue = int(result[6])
             nextTime = int(datetime.now().strftime('%H%M'))
 
-            # Check if caching is disabled
-            if (self.cache == False):
-                # Update the sensor state, divide by 100 to make it kWh
-                self._state = (nextValue / 100)
-                return
+            # Check if caching is enabled
+            if self.cache:
+                # Fetch data from the cache
+                try:
+                    cache = pickle.load(open(cacheName, 'rb'))
+                except (OSError, IOError, EOFError):
+                    cache = [0, 0]
 
-            # Fetch data from the cache
-            try:
-                cache = pickle.load(open(cacheName, 'rb'))
-            except (OSError, IOError, EOFError):
-                cache = [0, 0]
+                # Set the cache values
+                cacheValue = int(cache[0])
+                cacheTime = int(cache[1])
 
-            # Set the cache values
-            cacheValue = int(cache[0])
-            cacheTime = int(cache[1])
+                # If somehow the currentPowerToday is lower than the cached version,
+                # keep the cached version
+                if nextValue < cacheValue:
+                    nextValue = cacheValue
 
-            # If somehow the currentPowerToday is lower than the cached version,
-            # keep the cached version
-            if nextValue < cacheValue:
-                nextValue = cacheValue
+                # If today has passed, use the actual value from the Omnik inverter
+                if cacheTime > nextTime:
+                    nextValue = int(result[6])
 
-            # If today has passed, use the actual value from the Omnik inverter
-            if cacheTime > nextTime:
-                nextValue = int(result[6])
+                # Store new stats
+                pickle.dump([nextValue, nextTime], open(cacheName, 'wb'))
 
-            # Store new stats
-            pickle.dump([nextValue, nextTime], open(cacheName, 'wb'))
-
-            # Update the sensor state, divide by 100 to make it kWh
-            self._state = (nextValue / 100)
+                # Update the sensor state, divide by 100 to make it kWh
+                self._state = (nextValue / 100)
+            else:
+                # Update the sensor state, divide by 100 to make it kWh
+                self._state = (nextValue / 100)
```

Comment:
In that case, wouldn't this suffice as well? :)

```suggestion
            # Update the sensor state, divide by 100 to make it kWh
            self._state = (nextValue / 100)
```